### PR TITLE
th-230:  Eval order price on the BE

### PR DESCRIPTION
**Pull request by AlexVlaso**

### CODE REVIEW

**Review by MaxHopaytsa — APPROVED** [review]

File: backend/src/packages/map/map.service.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { type Client, type Distance, TravelMode } from './libs/types/types.js';
```
Comment (mV22110702) [inline backend/src/packages/map/map.service.ts]: TravelMode was used as enum. Should it be the part of types import, or should we make it as enum import instead?

File: backend/src/libs/packages/config/config.package.ts [hunk 2]
```diff
@@ -133,6 +133,14 @@ class Config implements IConfig {
           default: null,
         },
       },
+      API: {
+        GOOGLE_MAPS_API_KEY: {
+          doc: 'Key for Google maps API',
+          format: String,
+          env: 'VITE_APP_GOOGLE_MAPS_API_KEY',
```
Comment (PavloSukhinin) [inline backend/src/libs/packages/config/config.package.ts]: remove `VITE_APP` part - we don't use VITE on BE

File: backend/src/packages/map/map.service.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const response = await this.client.distancematrix({
+      params: {
+        key: this.key,
+        origins: [startPoint],
+        destinations: [endPoint],
+        mode: TravelMode.driving,
+      },
+    });
+
+    return response.data.rows[0].elements[0].distance;
```
Comment (PavloSukhinin) [inline backend/src/packages/map/map.service.ts]: avoid magic numbers

File: backend/src/packages/map/map.service.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    startPoint,
+    endPoint,
+    pricePerKm,
+  }: {
+    startPoint: string;
+    endPoint: string;
+    pricePerKm: number;
+  }): Promise<number> {
+    const distance = await this.getDistance(startPoint, endPoint);
+    const km = distance.value / 1000;
```
Comment (PavloSukhinin) [inline backend/src/packages/map/map.service.ts]: let's introduce a helper for it

**Review by PavloSukhinin — CHANGES_REQUESTED** [review]

**Review by mV22110702 — APPROVED** [review]

**Review by PavloSukhinin — DISMISSED** [review]

File: backend/src/packages/map/map.service.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    pricePerKm,
+  }: {
+    startPoint: string;
+    endPoint: string;
+    pricePerKm: number;
+  }): Promise<number> {
+    const distance = await this.getDistance(startPoint, endPoint);
+    const km = convertMetersToKm(distance.value);
+
+    return +(pricePerKm * km).toFixed(2);
```
Comment (kalinkaaaa14) [inline backend/src/packages/map/map.service.ts]:
```suggestion
const orderPrice = (pricePerKm * km).toFixed(2);

return Number(orderPrice);
```

File: backend/src/packages/map/libs/helpers/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+const convertMetersToKm = (meters: number): number => {
+  return meters / 1000;
```
Comment (kalinkaaaa14) [inline backend/src/packages/map/libs/helpers/helpers.ts]: let's create const METERS_IN_ONE_KM = 1000;

**Review by PavloSukhinin — APPROVED** [review]

**Review by kalinkaaaa14 — APPROVED** [review]

File: backend/src/packages/map/map.service.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        mode: TravelMode.driving,
+      },
+    });
+    const [resultArray] = response.data.rows;
+    const [resultElement] = resultArray.elements;
+
+    return resultElement.distance;
+  }
+
+  public async getPriceByDistance({
```
Comment (h0wter) [inline backend/src/packages/map/map.service.ts:33]: Is it correct to add price calculation logic in map service?

**Review by h0wter — APPROVED** [review]

**Pull request merged**